### PR TITLE
[RISCV][GISel] Add FCLASS to onlyUsesFP for register bank selection

### PR DESCRIPTION
**Pull request by lquinn2015**
Bug fix FCLASS instruction in RISCV. The bug is due the fact that FCLASS has an input  float register and output GPR this caused reg bank select regression.

### CODE REVIEW

File: llvm/utils/update_mir_test_checks.py [hunk 1]
```diff
@@ -87,13 +87,14 @@ def __call__(self, args, ir):
 
 
 class Run:
-    def __init__(self, prefixes, cmd_args, triple):
+    def __init__(self, prefixes, cmd_args, preprocess_cmd, triple):
```
Comment (topperc) [inline llvm/utils/update_mir_test_checks.py:90]: Can you split this into a separate commit?

Comment (lquinn2015) [reply llvm/utils/update_mir_test_checks.py:90]: Yeah no problem

File: llvm/test/CodeGen/RISCV/GlobalISel/regbankselect/fp-load-store.mir [hunk 2]
```diff
@@ -1,10 +1,10 @@
...
-# RUN:   -o - | FileCheck %s --check-prefixes=CHECK
-# RUN: llc -mtriple=riscv64 -mattr=+d,+zfh -run-pass=regbankselect \
-# RUN:   -disable-gisel-legality-check -simplify-mir -verify-machineinstrs %s \
-# RUN:   -o - | FileCheck %s --check-prefixes=CHECK
+# RUN: sed 's/XLen/32/g' %s | llc -mtriple=riscv32 -mattr=+d,+zfh -run-pass=regbankselect \
+# RUN:   -disable-gisel-legality-check -simplify-mir -verify-machineinstrs  -x mir \
+# RUN:   -o - | FileCheck %s --check-prefixes=CHECK-RV32
+# RUN: sed 's/XLen/64/g' %s | llc -mtriple=riscv64 -mattr=+d,+zfh -run-pass=regbankselect \
+# RUN:   -disable-gisel-legality-check -simplify-mir -verify-machineinstrs -x mir \
+# RUN:   -o - | FileCheck %s --check-prefixes=CHECK-RV64
```
Comment (arsenm) [inline llvm/test/CodeGen/RISCV/GlobalISel/regbankselect/fp-load-store.mir:7]: I think this is less maintainable than just duplicating the test. Does update_mir_test_checks even understand how to run the pre-sed command? It doesn't really parse and execute the command 

Also really should have end to end IR tests. Pure mir tests are overused at this point

Comment (lquinn2015) [reply llvm/test/CodeGen/RISCV/GlobalISel/regbankselect/fp-load-store.mir:7]: I would some what agree that spliting the test up would be more reasonable but that only case in the the new fp-load-store.mir test required that sed replacement was the one I added so it seemed sad to force duplication for just 1 part of the test case. On the other hand if we keep the tests conjoined it makes sure RV32 and RV64 cannot diverge on register select for these simple cases which is probably better imo.  @topperc asked me to keep them together maybe he had a different reason. 

You are correct that the update__mir_test_checks did not (past tense :)) handle preprocessor commands however I added that to the python script using the same infra that the update_llc_test_checks script uses so the two of them converged a little bit which I think is a plus.  I feel as though this is better as a MIR test because its directly testing the register select behavior that was changed. IR  would of course test that in a full sweep but it would also be indirectly testing a bunch of different behaviors that might change in the future.  Let me know what you think you guys are the bosses!

Comment (topperc) [reply llvm/test/CodeGen/RISCV/GlobalISel/regbankselect/fp-load-store.mir:7]: Let's move it to an IR test. I only suggested the MIR test because that's where we had existing coverage for the regbank selection for FP load/store.

**Comment by lquinn2015** [comment]
Cool so I dropped the .mir test and script change and added a float-fclass.ll  test

**Review by topperc — APPROVED** [review]
LGTM

**Review by topperc — APPROVED** [review]
LGTM

File: llvm/test/CodeGen/RISCV/GlobalISel/float-fclass.ll [hunk 3]
```diff
@@ -0,0 +1,21 @@
+; NOTE: Assertions have been autogenerated by utils/update_llc_test_checks.py UTC_ARGS: --version 5
+
+; RUN: llc -mtriple=riscv32 -mattr=+f -global-isel \
+; RUN:   -disable-gisel-legality-check -verify-machineinstrs \
```
Comment (topperc) [inline llvm/test/CodeGen/RISCV/GlobalISel/float-fclass.ll:4]: Can you drop the -disable-gisel-legality-check and -verify-machineinstrs.

Comment (lquinn2015) [reply llvm/test/CodeGen/RISCV/GlobalISel/float-fclass.ll]: Yeah no problem :)
